### PR TITLE
storage: multi-threads protection for PathPool

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/6002

Problem Summary:
`page_path_map` is not protected under some delegator methods
we need to add lock_guard everywhere we accessing `page_path_map`, which is error-prone

### What is changed and how it works?

* Protect `page_path_map` by lock in this PR
* Replace the `page_path_map` by a thread-safe wrapper class

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by JaySon-Huang** [comment]
pending for more unit tests

File: dbms/src/Storages/PathPool.cpp [hunk 1]
```diff
@@ -14,19 +14,22 @@
 
 #include <Common/Exception.h>
 #include <Common/Logger.h>
+#include <Common/SyncPoint/SyncPoint.h>
```
Comment (JaySon-Huang) [inline dbms/src/Storages/PathPool.cpp:17]:
```suggestion
```

**Comment by JaySon-Huang** [comment]
https://ci.pingcap.net/blue/organizations/jenkins/tiflash-sanitizer-daily/detail/tiflash-sanitizer-daily/555/pipeline/12

**Review by lidezhu — APPROVED** [review]
LGTM

**Review by hehechen — APPROVED** [review]

**Pull request merged**